src/Integration.Vsix/CFamily/CFamilyHelper.cs

```diff
@@ -117,60 +117,56 @@ private static Request CreateRequest(FileConfig fileConfig, string absoluteFileP
             return defaults;
         }
 
-        internal /* for testing */ static Response CallClangAnalyzer(Request request, IProcessRunner runner, ILogger logger, CancellationToken cancellationToken)
+        internal /* for testing */ static void CallClangAnalyzer(Action<Message> handleMessage, Request request, IProcessRunner runner, ILogger logger, CancellationToken cancellationToken)
         {
-            string tempFileName = null;
             try
             {
-                tempFileName = Path.GetTempFileName();
-
-                // Create a FileInfo object to set the file's attributes
-                FileInfo fileInfo = new FileInfo(tempFileName);
-
-                // Set the Attribute property of this file to Temporary. 
-                // Although this is not completely necessary, the .NET Framework is able 
-                // to optimize the use of Temporary files by keeping them cached in memory.
-                fileInfo.Attributes = FileAttributes.Temporary;
-
-                using (var writeStream = new FileStream(tempFileName, FileMode.Open))
+                if (analyzerExeFilePath == null)
                 {
-                    Protocol.Write(new BinaryWriter(writeStream), request);
+                    logger.WriteLine("Unable to locate the CFamily analyzer exe");
+                    return;
                 }
 
                 var workingDirectory = Path.GetTempPath();
-                var success = ExecuteAnalysis(runner, tempFileName, workingDirectory, logger, cancellationToken);
 
-                if ((request.Flags & Request.CreateReproducer) != 0)
+                var args = new ProcessRunnerArguments(analyzerExeFilePath, false)
                 {
-                    logger.WriteLine(CFamilyStrings.MSG_ReproducerSaved,
-                        Path.Combine(workingDirectory, "sonar-cfamily.reproducer"));
-
-                    // When running with reproducer flag, we don't want to show analysis results.
-                    return null;
-                }
-
-                if (success)
-                {
-                    using (var readStream = new FileStream(tempFileName, FileMode.Open))
+                    CmdLineArgs = new[] { "-" },
```

Comment:
Worth commenting that "-" tells the subprocess we want to communicate via standard IO streams.